internal/tools/aws/session.go

```diff
@@ -37,10 +37,10 @@ func NewAWSSessionWithLogger(config *aws.Config, logger log.FieldLogger) (*sessi
 				"aws-operation-name": r.Operation.Name,
 			})
 
-			if r.HTTPResponse.StatusCode >= 400 {
+			if r.HTTPResponse.StatusCode >= 500 {
```

Comment:
I've been meaning to address this, but in this part of the code, we should just always log `logger.Debug` because sometimes an error from AWS isn't an error in our application and logging `Error` here bubbles up the error as though it is an application-layer error. So I think we should get rid of these conditionals and just `logger.Debug(buffer.String())` here no matter what. What do you think?

Reply:
It wasn't supposed to be in the commit. I agree with having it only active for debugging. It was initially like that.